engine.py

```diff
@@ -23,200 +19,240 @@
+                for (panel_id, panel_dict) in self.panels.iteritems():
+                    nukescripts.panels.registerPanel(
+                        panel_id,
+                        panel_dict["callback"],
+                    )
+
+        # Iterate over all apps, if there is a gizmo folder, add it to nuke path.
         for app in self.apps.values():
-            # add gizmo to nuke path
+            # Add gizmos to nuke path.
             app_gizmo_folder = os.path.join(app.disk_location, "gizmos")
             if os.path.exists(app_gizmo_folder):
-                # now translate the path so that nuke is happy on windows
+                # Now translate the path so that nuke is happy on Windows.
                 app_gizmo_folder = app_gizmo_folder.replace(os.path.sep, "/")
                 self.log_debug("Gizmos found - Adding %s to nuke.pluginAddPath() and NUKE_PATH" % app_gizmo_folder)
                 nuke.pluginAddPath(app_gizmo_folder)
-                # and also add it to the plugin path - this is so that any 
+                # And also add it to the plugin path - this is so that any 
                 # new processes spawned from this one will have access too.
                 # (for example if you do file->open or file->new)
                 tank.util.append_path_to_env_var("NUKE_PATH", app_gizmo_folder)
-                   
+
     def destroy_engine(self):
         """
-        Runs when the engine is unloaded, e.g. typically at context switch.
+        Runs when the engine is unloaded, typically at context switch.
         """
         self.log_debug("%s: Destroying..." % self)
-        if self._ui_enabled:
+        if self.has_ui:
             self._menu_generator.destroy_menu()
-        
-
-    @property
-    def has_ui(self):
-        """
-        Detect and return if nuke is running in batch mode
-        """
-        return self._ui_enabled
 
     def _get_dialog_parent(self):
         """
-        Return the QWidget parent for all dialogs created through show_dialog & show_modal.
+        Return the QWidget parent for all dialogs created through
+        show_dialog and show_modal.
         """
-        # see https://github.com/shotgunsoftware/tk-nuke/commit/35ca540d152cc5357dc7e347b5efc728a3a89f4a 
-        # for more info. There have been instability issues with nuke 7 causing various crashes, so 
-        # window parenting on Nuke versions above 6 is currently disabled.
+        # See https://github.com/shotgunsoftware/tk-nuke/commit/35ca540d152cc5357dc7e347b5efc728a3a89f4a 
+        # for more info. There have been instability issues with nuke 7 causing
+        # various crashes, so window parenting on Nuke versions above 6 is
+        # currently disabled.
         if nuke.env.get("NukeVersionMajor") > 6:
             return None
         return super(NukeEngine, self)._get_dialog_parent()
 
-    ##########################################################################################
```

Comment:
I'm adding them back in now that I've also properly grouped together related methods/properties. The class is now easier to read!